### ISSUE TITLE
[BUG] Invalid stochastic rounding with infinite

### ISSUE THREAD

**Issue by Thukisdo** [milankl/StochasticRounding.jl#48]
Hi,

We've spotted an issue in `Float32_stochastic_round()` where rounding an Infinite will sometime return NaN, where we may expect Infinite, see output below. 

```
julia> Float32(Float32_stochastic_round(Inf))
Inf32

julia> Float32(Float32_stochastic_round(Inf))
NaN32
```

The only difference between NaN and infinite is an empty mantissa, hence adding a random noise will result in NaN.

**Comment by milankl** [comment]
Sorry @Thukisdo, missed that issue. Yes, I observed that earlier, but couldn't be bothered resolve this, as I didn't think this would be of practical importance. But maybe that's also me thinking that NaR (= one bitpattern for NaN, +-Inf) from posits is a good idea. I guess this could be resolved by another early exit, something like `isfinite(x) || return BFloat16sr(x)` such that for `x=NaN32` it returns `NaNB16sr` and for `x=Inf32` it returns `Inf16sr` ?

**Comment by Thukisdo** [comment]
No problem :)

Indeed, i also solved it using an early exit in my code.

**Comment by milankl** [comment]
Happy for you to create a PR with your solution?

**Comment by Thukisdo** [comment]
I applied this fix in a C++ adaptation of your rounding method [here](https://github.com/Thukisdo/NSan-interflop-runtime/blob/78d5ce142d1e52d81e72813041f56a43704b0f3e/src/backends/MCASync.cpp#L71), so i unfortunately do not have a patched Julia version.

**Comment by milankl** [comment]
Thanks @Thukisdo, this is addressed in upcoming v0.6.1 as I've replaced the stochastic rounding via signed integer arithmetic with unsigned integers. This avoids any stochastic rounding between Inf and NaN and doesn't require a check for special cases like `isfinite(x) || return ...` or `isnan(x) || return ...`. Check the new `Float32_stochastic_rounding` function definition if you are interested!
